openedx/core/djangoapps/user_api/accounts/tests/pact/user-middleware.py

```diff
@@ -0,0 +1,32 @@
+"""
+Contain the middleware logic needed during pact verification
+"""
+from django.contrib import auth
+from django.utils.deprecation import MiddlewareMixin
+
+User = auth.get_user_model()
+
+
+class AuthenticationMiddleware(MiddlewareMixin):
+    """
+    Middleware to add default authentication into the requests for pact verification.
+
+    This middleware is required to add a default authenticated user and bypass CSRF validation
+    into the requests during the pact verification workflow. Without the authentication, the pact verification
+    process will not work as the apis.
+    See https://docs.pact.io/faq#how-do-i-test-oauth-or-other-security-headers
+    """
+
+    def __init__(self, get_response):
+        super().__init__(get_response)
+        self.auth_user = User.objects.get_or_create(username='staff', is_staff=True)[0]
+        self.get_response = get_response
+
+    def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
+        """
+        Add a default authenticated user and remove CSRF checks for a request
+        in a subset of views.
+        """
+        if request.user.is_anonymous and 'Pact-Authentication' in request.headers:
```

Comment:
This enables some fairly dangerous looking behavior, so I wonder if there are additional safeguards we could add to ensure this is never accidentally enabled in production. A few options that come to mind:

- Introduce a Django setting called `PACT_BYPASS_AUTHENTICATION` that is set to True in the pact env and checked in this middleware's `__init__` method.
- Rename this middleware to `PactBypassAuthMiddleware` to better signal its purpose.
- Have the test user not have staff permission.

(Maybe some combination of these.)

Reply:
Is a `try-except `block for PACT_BYPASS_AUTHENTICATION`'s value in the `__init_` method sufficient? Also agree with your point on the renaming.

Reply:
One option would be to raise `django.core.exceptions.MiddlewareNotUsed` in the init, which will tell Django to quietly ignore the middleware. Or you could raise a regular Exception, for something louder; this might be appropriate, since the middleware should only be used in the `pact` environment, and the loud failure might help with debugging.

Reply:
Great, just added the `django.core.exceptions.MiddlewareNotUsed` exception if the bypass auth variable is false in latest commit.

Reply:
To clarify -- when I was suggesting renaming, I was referring to the class here, not the file. (And for file names we'll generally want to stick with `lower_snake_case` names.)